install.sh

```diff
@@ -147,30 +155,44 @@ setup_ramalama() {
   done
 }
 
-main() {
-  set -e -o pipefail
-
-  local local_install="false"
-  # Parse command line arguments
+parse_arguments() {
   while [[ $# -gt 0 ]]; do
     case "$1" in
       -l)
         local_install="true"
         shift
         ;;
       get_*)
-        get_installation_dir
+        get_install_dir="true"
         return;;
       *)
         break
     esac
   done
+}
+
+main() {
+  set -e -o pipefail
+
+  local local_install="false"
+  local get_install_dir="false"
+  parse_arguments "$@"
+  if $get_install_dir; then
+    get_installation_dir
+    return 0
+  fi
 
   local os
   os="$(uname -s)"
   local sudo=""
   check_platform
 
+  if available dnf; then
```

Comment:
Via dnf we'd get a versioned `ramalama` installation and with `setup_ramalama` we'd get the latest from main. So would it make sense to enable switching between these two options, e.g. with `--use-dnf`?

Reply:
This installs from the "s" branch by default which I push roughly whenever we do a release. I guess it would be nice to have alternate install techniques work even on systems that can install via dnf.

So I added checks to see if someone explicitly tried to specify a BRANCH or local install.

But we do want to install by dnf here by default. AFAIK this is the only install technique going to appear on the ramalama.ai website. And we want people on platforms like Fedora and RHEL to use the rpm version.

Reply:
This was always kinda meant as just a thrown together thing until the various distros package RamaLama properly. It will probably stay around forever because yeah it's nice to be able to install direct from upstream also like you say.